src/controllers/portfolio/portfolio.ts

```diff
@@ -138,6 +139,11 @@ export class PortfolioController extends EventEmitter {
       await this.#accounts.initialLoadPromise
       this.tokenPreferences = await this.#storage.get('tokenPreferences', [])
       this.#previousHints = await this.#storage.get('previousHints', {})
+      const networksWithAssets = await this.#storage.get('networksWithAssetsByAccount', {})
+
+      if (!Array.isArray(networksWithAssets)) {
+        this.#networksWithAssetsByAccounts = networksWithAssets
+      }
```

Comment:
This check is incorrect. `networksWithAssetsByAccount` was never an array- its values were. We should check every key's value or introduce a `@v2` storage key and no longer use the old one.

Reply:
![image](https://github.com/user-attachments/assets/9e7e7147-72d6-4603-a2c5-99f919a6ae3c)